src/libPMacc/include/algorithms/reverseBits.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2015 Alexander Grund
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include <boost/type_traits.hpp>
+#include <boost/static_assert.hpp>
+#include <climits>
+
+namespace PMacc {
+
+/**
+ * Reverses the bit in an unsigned integral value
+ *
+ * Based on "Bit Twiddling Hacks" by Sean Eron Anderson
+ * http://www.graphics.stanford.edu/~seander/bithacks.html (2015)
+ *
+ * @param value Value which should be reversed
+ * @return Reversed value
+ */
+template<typename T>
+T
+reverseBits(T value)
+{
+    BOOST_STATIC_ASSERT_MSG( boost::is_integral<T>::value && boost::is_unsigned<T>::value, "Only allowed for unsigned integral types" );
```

Comment:
please use `PMACC_CASSERT_MSG` because the currently used C++11 compiler write out no error message.